src/main/java/Parser.java

```diff
@@ -0,0 +1,127 @@
+import java.time.LocalDate;
+
+public class Parser {
+
+    public static boolean isBye(String command) {
+        return command.equalsIgnoreCase("bye");
+    }
+
+    public static boolean isList(String command) {
+        return command.equalsIgnoreCase("list");
+    }
+
+    public static boolean isMarkDone(String command) {
+        return command.startsWith("mark");
+    }
+
+    public static boolean isMarkNotDone(String command) {
+        return command.startsWith("unmark");
+    }
+
+    public static boolean isDelete(String command) {
+        return command.startsWith("delete");
+    }
+
+    public static int extractTaskIndex(String command) throws DukeException {
+        String[] parts = command.split(" ");
+        if (parts.length < 2) {
+            throw new DukeException("OOPS!!! Please provide the task index.");
+        }
+        return Integer.parseInt(parts[1].trim()) - 1;
+    }
+
+    public static Task parseTask(String command) throws DukeException {
+        String[] parts = command.split(" ", 2);
+
+        try {
+            TaskType taskType = TaskType.valueOf(parts[0].toUpperCase());
+
+            switch (taskType) {
+                case TODO:
+                    // Parse the description, fromDate, and toDate
+                    String[] todoParts = parts[1].split(" /from ");
```

Comment:
perhaps it would be good to split using strings without the black space characters followed by trimming the elements in todoParts when you want to use them? I find that it might be helpful if the user of the program accidentally enters multiple blankspace characters